lib/mutgen.c

```diff
@@ -433,21 +525,31 @@ mutgen_generate(mutgen_t *self, tsk_table_collection_t *tables, int flags)
         right = 0;
         while (right != edge_right) {
             right = GSL_MIN(edge_right, map_position[map_index + 1]);
-            mu = branch_length * (right - left) * map_rate[map_index];
+            site_left = discrete_sites ? ceil(left) : left;
```

Comment:
Not obvious to me while this is ceil and not floor. Is it necessary at all? Seems odd to change the "area" of the edges like this. Suppose you have left=0.99 and right=1.0001. Haven't you then inflated the probability of a mutation at 2 by taking the ceiling of both? I guess I'm not entirely clear what the model is when we combine floating point edge coords with integer sites.